airflow-core/src/airflow/ui/src/utils/tokenHandler.ts

```diff
@@ -36,7 +45,32 @@ const getTokenFromCookies = (): string | undefined => {
   return undefined;
 };
 
+const getRefreshedTokenFromCookies = (): string | undefined => {
+  const cookies = document.cookie.split(";");
+
+  for (const cookie of cookies) {
+    const [name, token] = cookie.split("=");
+
+    if (name?.trim() === "_refresh_token" && token !== undefined) {
+      setToken(token);
+      clearRefreshTokenCookie();
+
+      return token;
+    }
+  }
+
+  return undefined;
+};
+
 export const tokenHandler = (config: InternalAxiosRequestConfig) => {
+  const refreshToken = getRefreshedTokenFromCookies();
+
+  if (refreshToken !== undefined && localStorage.getItem(TOKEN_STORAGE_KEY) !== refreshToken) {
+    config.headers.Authorization = `Bearer ${refreshToken}`;
+
+    return config;
+  }
+
   const token = localStorage.getItem(TOKEN_STORAGE_KEY) ?? getTokenFromCookies();
```

Comment:
That should be the only change you need in the UI to make it work

```suggestion
  const token = getTokenFromCookies() ?? localStorage.getItem(TOKEN_STORAGE_KEY);
```

Reply:
Yes, especially after reverting the key changes